### DISCUSSION TITLE
To disable or not to disable

### DISCUSSION THREAD

**Discussion by frankhuurman** [zauberzeug/nicegui#2209] [General]
Just something funny I ran into while convinced that the correct argument to .props() was "disabled" to disable a checkbox, it's not! but it has interesting behaviour:

### Regular checkbox
```python
from nicegui import ui
ui.checkbox("Testing a checkbox")
ui.run()
```
![image](https://github.com/zauberzeug/nicegui/assets/19980966/3897df3d-3ca5-41cb-8ba3-e092144a059c)

### "Disabled" checkbox
```python
from nicegui import ui
ui.checkbox("Testing a checkbox").props("disabled")
ui.run()
```
![image](https://github.com/zauberzeug/nicegui/assets/19980966/c9f9c6ef-f431-403e-b740-a9f24fbdf4af)
![image](https://github.com/zauberzeug/nicegui/assets/19980966/af6f4da8-0894-497d-8438-34bbcd54fd4a)

As you can see, the checkbox is greyed out but it can still be toggled(also the cursor changes to a "disabled element" cursor)

### "Disable" checkbox
```python
from nicegui import ui
ui.checkbox("Testing a checkbox").props("disable")
ui.run()
```
![image](https://github.com/zauberzeug/nicegui/assets/19980966/1874fd47-9206-4850-8d42-bb11300878ca)

Just a slightly different grey tint, but this one really can't be toggled and is truly disabled.

Lesson of the day: 
Always make sure to read your Quasar docs to use the correct property name.

**Comment by falkoschindler** [comment]
Yes, this is confusing. See https://github.com/zauberzeug/nicegui/discussions/560 for a similar discussion, which led to PR https://github.com/zauberzeug/nicegui/pull/797. It introduced methods like `enable()` and `disable()` as well as a bindable property to provide a more official API for enabling/disabling elements.

**Reply by frankhuurman** [reply]
Haha slightly confusing but not a problem at all 👍 
Found it amusing to stumble upon a prop that did change something but wasn't the official prop to do it.

While on the topic of props, currently I'm binding visibility of one checkbox to another.
Is it possible to bind with enabling/disabling of props() as well?

For example how I'm doing it now:
```python
box_uno = ui.checkbox("Testy")
box_dos = ui.checkbox("Testy 2").bind_visibility(box_uno, "value")
```

I'm looking to enable the second element with something like this:
```python
box_uno = ui.checkbox("Testy")
box_dos = ui.checkbox("Testy 2").bind_value(box_uno, "disable")
# box_dos should be disabled by default until box_uno is checked
```

bind_enabled also doesn't seem to have an effect on the Quasar props, but perhaps I'm misreading things.

**Reply by falkoschindler** [reply]
Here you go:
```py
box_uno = ui.checkbox("Testy")
box_dos = ui.checkbox("Testy 2").bind_enabled_from(box_uno, 'value')
```

**Reply by frankhuurman** [reply]
Sometimes it feels so stupid having tried every combination of bindings except the correct one.

Thanks so much for the quick replies all the time Falko, real lifesaver most of the time.

I'm hoping we can show some thanks in the future by providing you guys with a badge that NiceGUI is proudly powering part of the Dutch government.

**Comment by stepheweffie** [comment]
I'm putting a list of links to read up. There's a lot of fun ways to hack this framework. I really enjoy it.
